### ISSUE TITLE
[sourcemap] ingesting apm test data doesn't yield transaction in apm ui

### ISSUE THREAD

**Issue by stuartnelson3** [elastic/apm-server#6947]
While manually testing https://github.com/elastic/apm-server/pull/6447 I uploaded a sourcemap + traces from the apm-server repo. The trace "appears" (ie. the ui acknowledges a trace has been ingested), but the specific trace isn't listed on the actual service ui.

uploading sourcemap:
```
curl -X POST \
  "http://admin:changeme@localhost:5601/api/apm/sourcemaps" \
  -H 'Content-Type: multipart/form-data' \
  -H 'kbn-xsrf: true' \
  -F 'service_name="apm-agent-js"' \
  -F 'service_version="1.0.0"' \
  -F 'bundle_filepath="http://localhost:8000/test/e2e/general-usecase/bundle.js.map"' \
  -F "sourcemap=$(jq '.|tojson' testdata/sourcemap/bundle.js.map)"
```
sending events:
```
curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/transactions_spans_rum_2.ndjson http://localhost:8200/intake/v2/rum/events
```
service dashboard:
![image](https://user-images.githubusercontent.com/1398104/147586062-fb2bf203-e2e7-4dd8-ad0a-88cb66e2c9b1.png)

From speaking with @Mpdreamz:

>It looks like the generated transaction metrics do not have `transaction.name` set.
>
>Inspecting the `get_service_transaction_groups` on the test data this appears to be a prerequisite for them to show up there.
>
>I don't see anything special in the rum data decoder for transaction.name either nor the transaction validate. We just validate that when its set it does not exceed 1024 characters.

Based on this, is this a bug, or expected?

**Comment by simitt** [comment]
Without going into too much details, but it looks like the sample data that you upload do not have a `transaction.name` set. Can you retry with this field being set initially?

**Comment by stuartnelson3** [comment]
works when I add `name` to the transaction. Should this be added to the sample data?

**Comment by simitt** [comment]
yes please add it to avoid further confusion and make testing easier.
